### PR TITLE
Support passing Dict[unicode, Any] in as **kwargs

### PR DESCRIPTION
**Pull request by rowillia**

### CODE REVIEW

File: test-data/unit/python2eval.test [hunk 1]
```diff
@@ -128,6 +128,21 @@ print '>', ['a', b'b', u'c']
 [out]
 > [u'a', 'b', u'c']
 
+[case testUnicodeLiteralsKwargs_python2]
```
Comment (gvanrossum) [inline test-data/unit/python2eval.test]: Can you also add a test that doesn't import unicode_literals but uses u'...' literals?

File: mypy/checkexpr.py [hunk 2]
```diff
@@ -1628,8 +1628,20 @@ def is_valid_var_arg(self, typ: Type) -> bool:
...
         """Is a type valid as a **kwargs argument?"""
-        return is_subtype(typ, self.chk.named_generic_type(
-            'builtins.dict', [self.named_type('builtins.str'), AnyType()]))
+        if self.chk.options.python_version[0] >= 3:
+            return is_subtype(typ, self.chk.named_generic_type(
+                'builtins.dict', [self.named_type('builtins.str'),
+                                  AnyType()]))
+        else:
+            return (
+                is_subtype(typ, self.chk.named_generic_type(
```
Comment (gvanrossum) [inline mypy/checkexpr.py]: Have you tried if it works with just the second condition (i.e. is_subtype(...dict[unicode]...))?

File: mypy/checkexpr.py [hunk 3]
```diff
@@ -1628,8 +1628,20 @@ def is_valid_var_arg(self, typ: Type) -> bool:
...
+                    'builtins.dict',
+                    [self.named_type('builtins.str'),
+                     AnyType()])) or
```
Comment (gvanrossum) [inline mypy/checkexpr.py]: I'd prefer not to hide this `or` behind such a long call. Maybe on a line by itself (if flake8 is okay with that) or at the start of the next line (which is now acceptable style in PEP 8)?

File: mypy/checkexpr.py (hunk 2 above)
Comment (rowillia) [inline mypy/checkexpr.py]: Unfortunately that didn't work (It broke the the test case with `str`).  I also tried

```py
UnionType.make_union([self.named_type('builtins.unicode'), self.named_type('builtins.str')])
```

and

```py
self.named_type('builtins.basestring')
```

File: test-data/unit/python2eval.test (hunk 1 above)
Comment (rowillia) [inline test-data/unit/python2eval.test]: Done

File: mypy/checkexpr.py (hunk 3 above)
Comment (rowillia) [inline mypy/checkexpr.py]: Done

**Comment by gvanrossum** [comment]
Thanks for the thorough research on alternatives! I am satisfied and have merged the PR and closed the issue.

**Pull request merged**